app/src/main/java/com/googleresearch/capturesync/SoftwareSyncController.java

```diff
@@ -249,21 +280,26 @@ private void setupSoftwareSync() {
             // Client.
             Map<Integer, RpcCallback> clientRpcs = new HashMap<>(sharedRpcs);
 
-            // Update status text to "waiting for leader".
+            // Adjust the state when waiting to connect to a leader.
             clientRpcs.put(
                     SyncConstants.METHOD_MSG_WAITING_FOR_LEADER,
-                    payload ->
-                            mMainActivity.runOnUiThread(
-                                    () -> mSyncStatus = mMainActivity.getString(R.string.rec_sync_waiting_for_leader, mSoftwareSync.getName())));
-
-            // Update status text to "waiting for sync".
+                    payload -> {
+                        mMainActivity.getApplicationInterface().getSoftwareSyncUtils().removeVideoRecordingPreparation();
+                        mIsVideoPreparationNeeded = false;
+                        mMainActivity.runOnUiThread(
+                                () -> mSyncStatus = mMainActivity.getString(R.string.rec_sync_waiting_for_leader, mSoftwareSync.getName()));
+                    });
+
+            // Adjust the state when establishing a connection to a leader.
             clientRpcs.put(
                     SyncConstants.METHOD_MSG_SYNCING,
-                    payload ->
-                            mMainActivity.runOnUiThread(
-                                    () -> mSyncStatus = mMainActivity.getString(R.string.rec_sync_waiting_for_sync, mSoftwareSync.getName())));
+                    payload -> {
+                        mIsVideoPreparationNeeded = false;
+                        mMainActivity.runOnUiThread(
+                                () -> mSyncStatus = mMainActivity.getString(R.string.rec_sync_waiting_for_sync, mSoftwareSync.getName()));
```

Comment:
Is it possible now to fully remove links to mainActivity from this controller?

Reply:
I don't think so as it is currently heavily used all over the controller. I can replace calls like to `getPreview()` and `getExtendedAppInterface()` with the corresponding fields, but the controller still will use `getString()` from `MainActivity` as a `Context`, `runOnUiThread()` from it as an `Activity`, `getResources()`, etc.

Reply:
All can be changed, for example, strings need only appcontext, UI thread launch can be done without activity at all (via handlers), but lets skip this now